### DISCUSSION TITLE
Compose: Renaming Op Codes

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#3203] [Development]
Currently, the Compose Dialog supports over 50 functions, e.g. `OP_COMPOSE_EDIT_TO`.
Almost all of them are named: `OP_COMPOSE_*`, but they act on different objects.

I'd like to rename them to the form: `OP_[OBJECT]_[ACTION]`

**Comment by flatcap** [comment]
# Altering an Email's **Envelope**

| Function              | Op Code                        | Rename to                       |
| :-------------------- | :----------------------------- | :------------------------------ |
| `<edit-bcc>`          | `OP_COMPOSE_EDIT_BCC`          | `OP_ENVELOPE_EDIT_BCC`          |
| `<edit-cc>`           | `OP_COMPOSE_EDIT_CC`           | `OP_ENVELOPE_EDIT_CC`           |
| `<edit-fcc>`          | `OP_COMPOSE_EDIT_FCC`          | `OP_ENVELOPE_EDIT_FCC`          |
| `<edit-followup-to>`  | `OP_COMPOSE_EDIT_FOLLOWUP_TO`  | `OP_ENVELOPE_EDIT_FOLLOWUP_TO`  |
| `<edit-from>`         | `OP_COMPOSE_EDIT_FROM`         | `OP_ENVELOPE_EDIT_FROM`         |
| `<edit-headers>`      | `OP_COMPOSE_EDIT_HEADERS`      | `OP_ENVELOPE_EDIT_HEADERS`      |
| `<edit-newsgroups>`   | `OP_COMPOSE_EDIT_NEWSGROUPS`   | `OP_ENVELOPE_EDIT_NEWSGROUPS`   |
| `<edit-reply-to>`     | `OP_COMPOSE_EDIT_REPLY_TO`     | `OP_ENVELOPE_EDIT_REPLY_TO`     |
| `<edit-subject>`      | `OP_COMPOSE_EDIT_SUBJECT`      | `OP_ENVELOPE_EDIT_SUBJECT`      |
| `<edit-to>`           | `OP_COMPOSE_EDIT_TO`           | `OP_ENVELOPE_EDIT_TO`           |
| `<edit-x-comment-to>` | `OP_COMPOSE_EDIT_X_COMMENT_TO` | `OP_ENVELOPE_EDIT_X_COMMENT_TO` |

**Reply by mschilli87** [reply]
I'd suggest to replace `ENV` with a full `ENVELOPE` to avoid potential confusion with 'environment'.

**Reply by flatcap** [reply]
Yeah
But, gonna need a wider monitor :-)

**Reply by mschilli87** [reply]
I myself do have apretty wide screen but tend to wrap lines around at the rather conservative length of 80 characters. Not sure how often these occur and you'll know best. I just figured I'd put it out there for you to consider. :wink:

**Comment by flatcap** [comment]
## Manipulating a set of **Attachments**

Note: `OP_DELETE` has been replaced with an Attachment-specific
`OP_ATTACHMENT_DETACH`, since the attached **file** isn't being deleted.

| Function                | Op Code                          | Rename to                           |
| :---------------------- | :------------------------------- | :---------------------------------- |
| `<attach-file>`         | `OP_COMPOSE_ATTACH_FILE`         | `OP_ATTACHMENT_ATTACH_FILE`         |
| `<attach-key>`          | `OP_COMPOSE_ATTACH_KEY`          | `OP_ATTACHMENT_ATTACH_KEY`          |
| `<attach-message>`      | `OP_COMPOSE_ATTACH_MESSAGE`      | `OP_ATTACHMENT_ATTACH_MESSAGE`      |
| `<attach-news-message>` | `OP_COMPOSE_ATTACH_NEWS_MESSAGE` | `OP_ATTACHMENT_ATTACH_NEWS_MESSAGE` |
| `<detach-file>`         | `OP_DELETE`                      | `OP_ATTACHMENT_DETACH`              |
| `<get-attachment>`      | `OP_COMPOSE_GET_ATTACHMENT`      | `OP_ATTACHMENT_GET_ATTACHMENT`      |
| `<group-alternatives>`  | `OP_COMPOSE_GROUP_ALTS`          | `OP_ATTACHMENT_GROUP_ALTS`          |
| `<group-multilingual>`  | `OP_COMPOSE_GROUP_LINGUAL`       | `OP_ATTACHMENT_GROUP_LINGUAL`       |
| `<group-related>`       | `OP_COMPOSE_GROUP_RELATED`       | `OP_ATTACHMENT_GROUP_RELATED`       |
| `<move-down>`           | `OP_COMPOSE_MOVE_DOWN`           | `OP_ATTACHMENT_MOVE_DOWN`           |
| `<move-up>`             | `OP_COMPOSE_MOVE_UP`             | `OP_ATTACHMENT_MOVE_UP`             |
| `<new-mime>`            | `OP_COMPOSE_NEW_MIME`            | `OP_ATTACHMENT_NEW_MIME`            |
| `<ungroup-attachment>`  | `OP_COMPOSE_UNGROUP_ATTACHMENT`  | `OP_ATTACHMENT_UNGROUP`             |

**Reply by mschilli87** [reply]
Also here, I'd rather spell out `ATTACHMENT` to make clear its about the object, not the action: `OP_ATTACHMENT_DETACH`  is pretty clear to me without context, while `OP_ATTACH_DETACH` could be confusing.

**Comment by flatcap** [comment]
## Altering **Attachment** Properties

| Function                    | Op Code                          | Rename to                          |
| :-------------------------- | :------------------------------- | :--------------------------------- |
| `<edit-content-id>`         | `OP_COMPOSE_EDIT_CONTENT_ID`     | `OP_ATTACHMENT_EDIT_CONTENT_ID`    |
| `<edit-description>`        | `OP_COMPOSE_EDIT_DESCRIPTION`    | `OP_ATTACHMENT_EDIT_DESCRIPTION`   |
| `<edit-encoding>`           | `OP_COMPOSE_EDIT_ENCODING`       | `OP_ATTACHMENT_EDIT_ENCODING`      |
| `<edit-language>`           | `OP_COMPOSE_EDIT_LANGUAGE`       | `OP_ATTACHMENT_EDIT_LANGUAGE`      |
| `<edit-mime>`               | `OP_COMPOSE_EDIT_MIME`           | `OP_ATTACHMENT_EDIT_MIME`          |
| `<edit-type>`               | `OP_EDIT_TYPE`                   | `OP_ATTACHMENT_EDIT_TYPE`          |
| `<rename-attachment>`       | `OP_COMPOSE_RENAME_ATTACHMENT`   | `OP_ATTACHMENT_RENAME_ATTACHMENT`  |
| `<toggle-disposition>`      | `OP_COMPOSE_TOGGLE_DISPOSITION`  | `OP_ATTACHMENT_TOGGLE_DISPOSITION` |
| `<toggle-recode>`           | `OP_COMPOSE_TOGGLE_RECODE`       | `OP_ATTACHMENT_TOGGLE_RECODE`      |
| `<toggle-unlink>`           | `OP_COMPOSE_TOGGLE_UNLINK`       | `OP_ATTACHMENT_TOGGLE_UNLINK`      |
| `<update-encoding>`         | `OP_COMPOSE_UPDATE_ENCODING`     | `OP_ATTACHMENT_UPDATE_ENCODING`    |

**Reply by mschilli87** [reply]
see https://github.com/neomutt/neomutt/discussions/3203#discussioncomment-2006554

**Comment by flatcap** [comment]
## Acting on an **Attachment**

These, I'm not sure about.
Should we split up the "generic" functions?

| Function         | Op Code     | Rename to              |
| :--------------- | :---------- | :--------------------- |
| `<copy-file>`    | `OP_SAVE`   | `OP_ATTACHMENT_SAVE`   |
| `<filter-entry>` | `OP_FILTER` | `OP_ATTACHMENT_FILTER` |
| `<pipe-entry>`   | `OP_PIPE`   | `OP_ATTACHMENT_PIPE`   |
| `<print-entry>`  | `OP_PRINT`  | `OP_ATTACHMENT_PRINT`  |

**Reply by mschilli87** [reply]
see https://github.com/neomutt/neomutt/discussions/3203#discussioncomment-2006554
